main.nf

```diff
@@ -0,0 +1,708 @@
+      --fasta                       Path to Fasta reference
+
+    Other options:
+      --outdir                      The output directory where the results will be saved
+      --email                       Set this parameter to your e-mail address to get a summary e-mail with details of the run sent to you when the workflow exits
+      -name                         Name for the pipeline run. If not specified, Nextflow will automatically generate a random mnemonic.
+
+    AWSBatch options:
+      --awsqueue                    The AWSBatch JobQueue that needs to be set when running on AWSBatch
+      --awsregion                   The AWS Region for your AWS Batch job to run on
+    """.stripIndent()
+}
+
+/*
+ * SET UP CONFIGURATION VARIABLES
+ */
+
+// Show help emssage
+if (params.help){
+    helpMessage()
+    exit 0
+}
+
+// Configurable variables
+params.name = false
+params.test = false
+params.fasta = params.genome ? params.genomes[ params.genome ].fasta ?: false : false
+params.gtf = params.genome ? params.genomes[ params.genome ].gtf ?: false : false
+params.star_index = params.genome ? params.genomes[ params.genome ].star ?: false : false
+params.multiqc_config = "$baseDir/conf/multiqc_config.yaml"
+params.email = false
+params.plaintext_email = false
+
+multiqc_config = file(params.multiqc_config)
+output_docs = file("$baseDir/docs/output.md")
+
+// AWSBatch sanity checking
+if(workflow.profile == 'awsbatch'){
+    if (!params.awsqueue || !params.awsregion) exit 1, "Specify correct --awsqueue and --awsregion parameters on AWSBatch!"
+    if (!workflow.workDir.startsWith('s3') || !params.outdir.startsWith('s3')) exit 1, "Specify S3 URLs for workDir and outdir parameters on AWSBatch!"
+}
+
+// Has the run name been specified by the user?
+//  this has the bonus effect of catching both -name and --name
+custom_runName = params.name
+if( !(workflow.runName ==~ /[a-z]+_[a-z]+/) ){
+  custom_runName = workflow.runName
+}
+
+// Check workDir/outdir paths to be S3 buckets if running on AWSBatch
+// related: https://github.com/nextflow-io/nextflow/issues/813
+if( workflow.profile == 'awsbatch') {
+    if(!workflow.workDir.startsWith('s3:') || !params.outdir.startsWith('s3:')) exit 1, "Workdir or Outdir not on S3 - specify S3 Buckets for each to run on AWSBatch!"
+}
+
+// Validate pipeline variables
+// These variable have to be defined in the profile configuration which is referenced in nextflow.config
+if (!params.genome) {
+    exit 1, "Mandatory parameter genome not specified!"    
+}
+
+if (params.fasta) {
+    fasta = file(params.fasta)
+    if(!fasta.exists()) exit 1, "Fasta file not found: ${params.fasta}"
+}
+
+if (params.gtf) {
+    gtf = file(params.gtf)
+    if(!gtf.exists()) exit 1, "GTF file not found: ${params.fasta}"
+}
+
+if(params.star_index){
+    star_index = Channel
+        .fromPath(params.star_index)
+        .ifEmpty { exit 1, "STAR index not found: ${params.star_index}" }
+}
+
+star_fusion_ref = ''
```

Comment:
Do you need to declare this channel beforehand?

Is the `star_fusion_ref = Channel...` not enough?

Reply:
I need to define somewhere the variable because I am calling a `star_fusion_ref.collect()`. Moved all variables (up)[https://github.com/nf-core/rnafusion/blob/f2af8d92420b0a24f8564c77867bae26679f3376/main.nf#L76].